tests/modules/moduleconfigs.c

```diff
@@ -0,0 +1,139 @@
+        *err = RedisModule_CreateString(NULL, "Bool configs cannot both be yes.", 32);
+        return REDISMODULE_ERR;
+    }
+    return REDISMODULE_OK;
+}
+
+int longlongApplyFunc(RedisModuleCtx *ctx, void *privdata, RedisModuleString **err) {
+    REDISMODULE_NOT_USED(ctx);
+    REDISMODULE_NOT_USED(privdata);
+    if (longval == memval) {
+        *err = RedisModule_CreateString(NULL, "These configs cannot equal each other.", 38);
+        return REDISMODULE_ERR;
+    }
+    return REDISMODULE_OK;
+}
+
+int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    REDISMODULE_NOT_USED(argv);
+    REDISMODULE_NOT_USED(argc);
+
+    if (RedisModule_Init(ctx, "moduleconfigs", 1, REDISMODULE_APIVER_1) == REDISMODULE_ERR) return REDISMODULE_ERR;
+
+    if (RedisModule_RegisterBoolConfig(ctx, "mutable_bool", 1, REDISMODULE_CONFIG_DEFAULT, getBoolConfigCommand, setBoolConfigCommand, boolApplyFunc, &mutable_bool_val) == REDISMODULE_ERR) {
+        return REDISMODULE_ERR;
+    }
+    /* Immutable config here. */
+    if (RedisModule_RegisterBoolConfig(ctx, "immutable_bool", 0, REDISMODULE_CONFIG_IMMUTABLE, getBoolConfigCommand, setBoolConfigCommand, boolApplyFunc, &immutable_bool_val) == REDISMODULE_ERR) {
+        return REDISMODULE_ERR;
+    }
+    if (RedisModule_RegisterStringConfig(ctx, "string", "secret password", REDISMODULE_CONFIG_DEFAULT, getStringConfigCommand, setStringConfigCommand, NULL, NULL) == REDISMODULE_ERR) {
+        return REDISMODULE_ERR;
+    }
+
+    /* On the stack to make sure we're copying them. */
+    const char *enum_vals[3] = {"one", "two", "three"};
+    const int int_vals[3] = {0, 2, 4};
+
+    if (RedisModule_RegisterEnumConfig(ctx, "enum", 0, REDISMODULE_CONFIG_DEFAULT, enum_vals, int_vals, 3, getEnumConfigCommand, setEnumConfigCommand, NULL, NULL) == REDISMODULE_ERR) {
+        return REDISMODULE_ERR;
+    }
+    /* Memory config here. */
+    if (RedisModule_RegisterNumericConfig(ctx, "memory_numeric", 1024, REDISMODULE_CONFIG_DEFAULT | REDISMODULE_CONFIG_MEMORY, 0, 3000000, getNumericConfigCommand, setNumericConfigCommand, longlongApplyFunc, &memval) == REDISMODULE_ERR) {
+        return REDISMODULE_ERR;
+    }
+    if (RedisModule_RegisterNumericConfig(ctx, "numeric", -1, REDISMODULE_CONFIG_DEFAULT, -5, 2000, getNumericConfigCommand, setNumericConfigCommand, longlongApplyFunc, &longval) == REDISMODULE_ERR) {
+        return REDISMODULE_ERR;
+    }
+    size_t len;
+    if (argc && !strcasecmp(RedisModule_StringPtrLen(argv[0], &len), "noload")) {
+        return REDISMODULE_OK;
+    } else if (RedisModule_LoadConfigs(ctx) == REDISMODULE_ERR) {
+        if (strval) {
+            RedisModule_FreeString(ctx, strval);
+            strval = NULL;
+        }
+        return REDISMODULE_ERR;
+    }
+    return REDISMODULE_OK;
+}
+
+int RedisModule_OnUnload(RedisModuleCtx *ctx) {
+    REDISMODULE_NOT_USED(ctx);
+    if (strval) RedisModule_FreeString(ctx, strval);
```

Comment:
I reproduced the fail test on alpine, when I comment out this line test is OK.
But I didn't find out the real reason why, no stack on alpine is too painful.

Reply:
The only thing I can think of is that on alpine the dynamic library loader doesn't properly in its the global (maybe when the module is unloaded and reloaded). 
@sundb can you try to init that global to null from the `onload` function and see if it helps?

Reply:
@oranagra  You guessed it right. strval needs to be reset to NULL after free.
Change code to following will be OK:

```
    if (strval) {
        RedisModule_FreeString(ctx, strval);
        strval = NULL;
    }
```

Reply:
Please make a PR to fix both issues.